android/src/main/kotlin/io/customer/customer_io/CustomerIoPlugin.kt

```diff
@@ -142,9 +154,28 @@ class CustomerIoPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
         }
     }
 
+    private fun registerDeviceToken(params: Map<String, Any>) {
+        val token = params.getString(Keys.Tracking.TOKEN)
+        CustomerIO.instance().registerDeviceToken(token)
+    }
+
+    private fun trackMetric(params: Map<String, Any>) {
+        val deliveryId = params.getString(Keys.Tracking.DELIVERY_ID)
+        val deliveryToken = params.getString(Keys.Tracking.DELIVERY_TOKEN)
+        val event = MetricEvent.getEvent(params.getProperty<String>(Keys.Tracking.METRIC_EVENT))
+
+        if (event == null) {
+            logger.info("metric event type isn't correct")
```

Comment:
Can we move this logic of validating the metric event type into the dart code? 

I understand that there is a chance that `MetricEvent.getEvent()` can still return nil in this Kotlin code layer, but if it does, I think it would be considered an SDK bug instead of an error for the customer. Therefore, I would expect to see the log statement be `logger.error("metric event null. Possible issue with SDK code? Given: ${params.getProperty<String>(Keys.Tracking.METRIC_EVENT)}")`

Reply:
We do perform the validation in the `dart` file and are certain that the `event` sent will be catered by the SDK. 

I can update the log statement.